### PR TITLE
resolve: several mdns fixlets

### LINKED ISSUE #29168: mDNS queries not working with known answers appended.
### systemd version the issue has been seen with

252

### Used distribution

yocto open embedded 

### Linux kernel version used

5.4.234

### CPU architectures issue was seen on

arm

### Component

systemd-resolved

### Expected behaviour you didn't see

mDNS quires with known answer appended should be processed without any error.

### Unexpected behaviour you saw
The issue is systemd mdns publisher/resolver rejecting these queries appended with known answers with error "Received mDNS query without question, ignoring".

### Steps to reproduce the problem

Use two devices one as systemd querier (Device A) and other one as systemd mdns publisher/resolver (Device B).

1.Enable systemd mDNS on both devices

2.Add dnssd service with name "_test._udp.local" in Device B

3.Restart systemd services on both devices to make sure caches are flushed and dnssd service is populated.

4.Issue the 1st query from Device A
resolvectl query --protocol=mdns --interface=eth0 --type=PTR _test._udp.local

5.Issue the 2nd query from Device A **before TTL expired in 2mintues with cache disabled flag.**
resolvectl query --protocol=mdns --interface=eth0 --cache=false --type=PTR _test._udp.local

### Additional program output to the terminal or log subsystem illustrating the issue

Observation from tcpdump 
1.Truncked flag is enabled even when a known answer is appended successfully in 1st packet (Packet No#3 in tcpdump).

image

2.A subsequent empty query without question and answer sent from systemd querier. (Packet No#4 in tcpdump).
image

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #29168.

### CODE REVIEW

**Comment by yuwata** [comment]
@mrc0mmand IIRC, we do not have any tests for mDNS. Do you have any idea about that?

File: src/resolve/resolved-manager.c [hunk 1]
```diff
@@ -1198,8 +1198,11 @@ int manager_send(
         assert(port > 0);
         assert(p);
 
+        /* For mDNS, it is natural that the packet have truncated flag when we have much known answers. */
```
Comment (poettering) [inline src/resolve/resolved-manager.c]: much → many

**Comment by mrc0mmand** [comment]
> @mrc0mmand IIRC, we do not have any tests for mDNS. Do you have any idea about that?

There's a couple of mDNS tests in TEST-75-RESOLVED, though I'm not sure how in-depth they are.

**Comment by yuwata** [comment]
> > @mrc0mmand IIRC, we do not have any tests for mDNS. Do you have any idea about that?
> 
> There's a couple of mDNS tests in TEST-75-RESOLVED, though I'm not sure how in-depth they are.

I think that's only about setting enable/disable/resolve. Sending query or so is not tested, IIRC.

**Pull request merged**